pkg/scalers/azure_monitor_scaler.go

```diff
@@ -0,0 +1,211 @@
+package scalers
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	v2beta1 "k8s.io/api/autoscaling/v2beta1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/metrics/pkg/apis/external_metrics"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	azureMonitorMetricName       = "metricName"
+	targetValueName              = "targetValue"
+	defaultSubscriptionIDSetting = "xxx"
+	defaultTenantIDSetting       = "yyy"
+	defaultClientIDSetting       = "zzz"
+	defaultClientPasswordSetting = "qqq"
+)
+
+type azureMonitorScaler struct {
+	metadata *azureMonitorMetadata
+}
+
+type azureMonitorMetadata struct {
+	resourceURI         string
+	tenantID            string
+	subscriptionID      string
+	resourceGroupName   string
+	name                string
+	filter              string
+	aggregationInterval string
+	aggregationType     string
+	clientID            string
+	clientPassword      string
+	targetValue         int
+}
+
+var azureMonitorLog = logf.Log.WithName("azure_monitor_scaler")
+
+// NewAzureMonitorScaler stuff
+func NewAzureMonitorScaler(resolvedEnv, metadata, authParams map[string]string) (Scaler, error) {
+	meta, err := parseAzureMonitorMetadata(metadata, resolvedEnv, authParams)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing azure monitor metadata: %s", err)
+	}
+
+	return &azureMonitorScaler{
+		metadata: meta,
+	}, nil
+}
+
+func parseAzureMonitorMetadata(metadata, resolvedEnv, authParams map[string]string) (*azureMonitorMetadata, error) {
+	meta := azureMonitorMetadata{}
+
+	if val, ok := metadata[targetValueName]; ok && val != "" {
+		targetValue, err := strconv.Atoi(val)
+		if err != nil {
+			azureMonitorLog.Error(err, "Error parsing azure monitor metadata", "targetValue", targetValueName)
+			return nil, fmt.Errorf("Error parsing azure monitor metadata %s: %s", targetValueName, err.Error())
+		}
+		meta.targetValue = targetValue
+	}
+
+	if val, ok := metadata["resourceURI"]; ok && val != "" {
+		meta.resourceURI = val
+	} else {
+		return nil, fmt.Errorf("no resourceURI given")
+	}
+
+	if val, ok := metadata["resourceGroupName"]; ok && val != "" {
+		meta.resourceGroupName = val
+	} else {
+		return nil, fmt.Errorf("no resourceGroupName given")
+	}
+
+	if val, ok := metadata[azureMonitorMetricName]; ok && val != "" {
+		meta.name = val
+	} else {
+		return nil, fmt.Errorf("no metricName given")
+	}
+
+	if val, ok := metadata["metricAggregationType"]; ok && val != "" {
```

Comment:
`formatTimeSpan` returns a 5 minute time interval if no `aggregationInterval` is provided. It might be a good idea to name it something more descriptive? Or have two separate ones? Like `formatTimeSpan` and `getDefaultOrCustomTimeSpan`?

Reply:
I think you can leave it as is, just make sure that default is documented (you may have already done this)